use-cases.md

```diff
@@ -69,11 +65,7 @@ The `operation` above refers to the name the server wants to give to the Span. F
 The `span` object above can be `None` if the Tracer did not find relevant headers in the incoming request: presumably because the client did not send them. In this case the server needs to start a brand new trace.
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
>  end user cannot write any Extractors, the encoding details are private to the tracer.

Yes and No.

The end user can write Injector/Extractors, but yes they need to honour existing OT protocol-specific specification. This is tackled by providing base Injectors/Extractors for protocols and protocol-specific specifications. I don't think we should be bloating the primary OT Specification with protocol specifics, but rather provide separate Protocol Specifications. This too creates the clean separation between Public API and Internal APIs.